engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -357,10 +366,18 @@ public void onCalledProcessCompleted(
       final BpmnElementContext childContext, final BpmnElementContext parentInstanceContext) {
     final var containerScope = getParentProcessScope(parentInstanceContext, childContext);
     final var containerProcessor = processorLookUp.apply(containerScope.getElementType());
-    containerProcessor.beforeExecutionPathCompleted(
-        containerScope, parentInstanceContext, childContext);
+    final var satisfiesCompletionCondition = new AtomicBoolean();
+    containerProcessor
+        .beforeExecutionPathCompleted(containerScope, parentInstanceContext, childContext)
+        .map(
+            ok -> {
+              if (ok instanceof Boolean) {
+                satisfiesCompletionCondition.set((Boolean) ok);
+              }
+              return null;
+            });
     containerProcessor.afterExecutionPathCompleted(
-        containerScope, parentInstanceContext, childContext);
+        containerScope, parentInstanceContext, childContext, satisfiesCompletionCondition.get());
```

Comment:
🔧 Similar as above, we can clean this up (and it highlights that we should deal with the left case 😅)
```java
final Either<Failure, ?> satisfiesCompletionConditionOrFailure =
    containerProcessor.beforeExecutionPathCompleted(
        containerScope, parentInstanceContext, childContext);
// todo(@korthout): deal with the left case of satisfiesCompletionConditionOrFailure
satisfiesCompletionConditionOrFailure.ifRight(
    satisfiesCompletionCondition ->
        containerProcessor.afterExecutionPathCompleted(
            containerScope, parentInstanceContext, childContext, satisfiesCompletionCondition));
```

Reply:
Hi @korthout , should i need to raise an incident for the left case?

Reply:
No that's definitely out of scope. TBH, let's leave the wrong implementation there the same as it was:
```
final Either<Failure, ?> satisfiesCompletionConditionOrFailure =
    containerProcessor.beforeExecutionPathCompleted(
        containerScope, parentInstanceContext, childContext);
// todo(@korthout): deal with the left case of satisfiesCompletionConditionOrFailure
containerProcessor.afterExecutionPathCompleted(
    containerScope, parentInstanceContext, childContext, satisfiesCompletionConditionOrFailure.get()));
```

I'll open a bug for this